tools/pythonpkg/src/pyconnection.cpp

```diff
@@ -150,6 +157,47 @@ static void InitializeConnectionMethods(py::class_<DuckDBPyConnection, shared_pt
 	    .def("load_extension", &DuckDBPyConnection::LoadExtension, "Load an installed extension", py::arg("extension"));
 }
 
+void DuckDBPyConnection::UnregisterFilesystem(const py::str &name) {
+	auto &fs = database->GetFileSystem();
+
+	fs.UnregisterSubSystem(name);
+}
+
+void DuckDBPyConnection::RegisterFilesystem(const AbstractFileSystem &filesystem) {
+	PythonGILWrapper gil_wrapper;
+
+	if (!py::isinstance<AbstractFileSystem>(filesystem)) {
+		throw InvalidInputException("Bad filesystem instance");
+	}
+
+	auto &fs = database->GetFileSystem();
+
+	auto protocol = filesystem.attr("protocol");
+	if (protocol.is_none() || py::str("abstract").equal(protocol)) {
+		throw InvalidInputException("Must provide concrete fsspec implementation");
+	}
+
+	vector<string> protocols;
+	if (py::isinstance<py::str>(protocol)) {
+		protocols.push_back(py::str(protocol));
+	} else {
+		for (const auto &sub_protocol : protocol) {
+			protocols.push_back(py::str(sub_protocol));
+		}
+	}
+
+	fs.RegisterSubSystem(make_unique<PythonFilesystem>(protocols, filesystem));
```

Comment:
Sorry I wasn't clear earlier
Can you move the vector into this?
My first comment was about avoiding needlessly copying the vector.

But to pass it as a reference requires the vector to be saved by something else.
Passing a reference to this vector is a problem because it's only temporary and will go out of scope